frappe/core/doctype/user/user.py

```diff
@@ -363,7 +363,7 @@ def get_timezones():
 def get_all_roles(arg=None):
 	"""return all roles"""
 	return [r[0] for r in frappe.db.sql("""select name from tabRole
-		where name not in ('Administrator', 'Guest', 'All') order by name""")]
+		where name not in ('Administrator', 'Guest', 'All') and not disabled order by name""")]
```

Comment:
Did you test this? I am not sure mysql allows `not disabled`. I think it should be `and disabled=0`